python/src/scipp/compat/mantid.py

```diff
@@ -4,13 +4,45 @@
 
 import re
 from copy import deepcopy
+from contextlib import contextmanager
 
 import numpy as np
 
 from .. import detail
 from .._scipp import core as sc
 
 
+@contextmanager
+def run_mantid_alg(alg, *args, **kwargs):
+    try:
+        from mantid import simpleapi as mantid
+        from mantid.api import AnalysisDataService
+    except ImportError:
+        raise ImportError(
+            "Mantid Python API was not found, please install Mantid framework "
+            "as detailed in the installation instructions (https://scipp."
+            "readthedocs.io/en/latest/getting-started/installation.html)")
+    # Deal with multiple calls to this function, which may have conflicting
+    # names in the global AnalysisDataService.
+    run_mantid_alg.workspace_id += 1
+    ws_name = f'scipp.run_mantid_alg.{run_mantid_alg.workspace_id}'
+    # Deal with non-standard ways to define the prefix of output workspaces
+    if alg == 'Fit':
+        kwargs['Output'] = ws_name
+    else:
+        kwargs['OutputWorkspace'] = ws_name
+    ws = getattr(mantid, alg)(*args, **kwargs)
```

Comment:
Did you consider trying to create an unmanaged algorithm https://docs.mantidproject.org/nightly/api/python/mantid/api/AlgorithmManagerImpl.html, via `createUnmanaged` then configure the algorithm manually (`setProperty`)? I'm not sure if that solves the underlying cause but it might if you suspect the proxy or simplapi wrapper to be getting in the way.

Reply:
Yes I did, but setting properties in a generic way was not trivial. We would still need to delete the workspaces manually though, I think.

Reply:
Well, probably not having the algorithm managed might help, but note that not all algorithms (like `Fit`) support the `StoreInADS` option.

Reply:
But you are probably right, I should try again with the algorithm manager. The current way still has issues with some algorithms that do not support the `OutputWorkspace` property.

Reply:
@OwenArnold Trying again, I remember whay was my issue with using `AlgorithmManager`: It does not return the created outputs (everything is done via properties), so I could not see a way to extract those in a generic way.

Reply:
Ok. In that case I'll just approve as is.

Reply:
I will actually need to push and update, for yet another special way of defining the output workspace name. :(

Reply:
Would `getProperies` on a generated Algorithm help? you can then go through and find names of Output properties without guessing.

Reply:
I suppose that would essentially reimplement what is done in `simpleapi`?

Reply:
Yes, and it's not going to be fully helpful with multiple output properies for example. But in the case of single output property prevents the need to switch on the specific algorithm to bind ws_name to output. That's just a minor improvement, I'm happy for this to be merged as is.